Example 2 for Butterfly chart (version2)

SimaRaha opened the pull request:
Please uncomment this block and take a look at this checklist if your PR is making substantial changes to **documentation**/impacts files in the `doc` directory. Check all that apply to your PR, and leave the rest unchecked to discuss with your reviewer! Not all boxes must be checked for every PR :)

If your PR modifies code of the `plotly` package, we have a different checklist
below :-).

### Documentation PR

- [ ] I've [seen the `doc/README.md` file](https://github.com/plotly/plotly.py/blob/master/doc/README.md)
- [ Yes] This change runs in the current version of Plotly on PyPI and targets the `doc-prod` branch OR it targets the `master` branch
- [ N/A instead used plotly.graph_objects ] If this PR modifies the first example in a page or adds a new one, it is a `px` example if at all possible
- [ ] Every new/modified example has a descriptive title and motivating sentence or paragraph
- [ Yes] Every new/modified example is independently runnable
- [ Yes] Every new/modified example is optimized for short line count	and focuses on the Plotly/visualization-related aspects of the example rather than the computation required to produce the data being visualized
- [ ] Meaningful/relatable datasets are used for all new examples instead of randomly-generated data where possible
- [Yes ] The random seed is set if using randomly-generated data in new/modified examples
- [ Yes] New/modified remote datasets are loaded from https://plotly.github.io/datasets and added to https://github.com/plotly/datasets
- [ ] Large computations are avoided in the new/modified examples in favour of loading remote datasets that represent the output of such computations
- [Yes ] Imports are `plotly.graph_objects as go` / `plotly.express as px` / `plotly.io as pio`
- [ ] Data frames are always called `df`
- [ Yes] `fig = <something>` call is high up in each new/modified example (either `px.<something>` or `make_subplots` or `go.Figure`)
- [Yes ] Liberal use is made of `fig.add_*` and `fig.update_*` rather than `go.Figure(data=..., layout=...)` in every new/modified example
- [ Yes] Specific adders and updaters like `fig.add_shape` and `fig.update_xaxes` are used instead of big `fig.update_layout` calls in every new/modified example
- [ Yes] `fig.show()` is at the end of each new/modified example
- [ Yes] `plotly.plot()` and `plotly.iplot()` are not used in any new/modified example
- [ Yes] Hex codes for colors are not used in any new/modified example in favour of [these nice ones](https://github.com/plotly/plotly.py/issues/2192)

## Code PR

- [ ] I have read through the [contributing notes](https://github.com/plotly/plotly.py/blob/master/contributing.md) and understand the structure of the package. In particular, if my PR modifies code of `plotly.graph_objects`, my modifications concern the `codegen` files and not generated files.
- [ ] I have added tests (if submitting a new feature or correcting a bug) or
  modified existing tests.
- [ ] For a new feature, I have added documentation examples in an existing or
  new tutorial notebook (please see the doc checklist as well).
- [ ] I have added a CHANGELOG entry if fixing/changing/adding anything substantial.
- [ ] For a new feature or a change in behaviour, I have updated the relevant docstrings in the code to describe the feature or behaviour (please see the doc checklist as well).

-->

SimaRaha committed:
Added plotly-universities-1.ipynb for visualization updates

SimaRaha commented:
Hi Greg and Liam,

 I've converted this pull request to a draft as we're still working on finalizing some aspects. I'll let you know once it's ready for review.

rl-utility-man commented:
This is a continuation of #4994

rl-utility-man committed:
butterfly with neutral column

rl-utility-man committed:
Delete scratchwork file

rl-utility-man committed:
replaced hard coded data with a read_csv from github

gvwilson commented:
@LiamConnors can you please review for the Plotly.py 6.1 release?

rl-utility-man committed:
improved the comments.

rl-utility-man committed:
add python decorator

LiamConnors commented on doc/python/horizontal-bar-charts.md:
```diff
@@ -214,6 +214,108 @@ for yd, xd in zip(y_data, x_data):
+        name=col,
+        orientation='h',
+        marker=dict(color=color_by_category[col]),
+        legendrank=legend_rank_by_category[col],
+        xaxis=f"x{1+(col=="Neither Agree nor Disagree")}", # in this context, putting "Neither Agree nor Disagree" on a secondary x-axis on a different domain 
+                       # yields results equivalent to subplots with far less code
```
The example doesn't run for me, and I believe it's related to setting of the xaxis here.

rl-utility-man replied:
Good point.  I had " where I needed ' .  Fixed.

rl-utility-man commented on doc/python/horizontal-bar-charts.md:
```diff
@@ -214,6 +214,108 @@ for yd, xd in zip(y_data, x_data):
+import plotly.graph_objects as go
+
+
+df = pd.read_csv('https://raw.githubusercontent.com/plotly/datasets/refs/heads/master/gss_2002_5_pt_likert.csv')
+df.rename(columns={'Unnamed: 0':"Category"}, inplace=True)
```
It's a shortcoming of the data set.  I just proposed a PR to label the column properly in the data set https://github.com/plotly/datasets/pull/64  a search of github shows no uses of that data set other than in this PR and #4994, so it appears safe to accept that PR.  (I uploaded this data set recently in https://github.com/plotly/datasets/pull/62 )

rl-utility-man committed:
corrected quotation mark type

rl-utility-man committed:
replace an update_layout call with a go.figure() call

rl-utility-man committed:
moved yaxis title and legned layout to go.Figure()

rl-utility-man committed:
introductory text edit.

rl-utility-man committed:
attempted to manually merge the latest changes to doc-prod so the new contribution is the second butterfly example

rl-utility-man commented on doc/python/horizontal-bar-charts.md:
````diff
@@ -217,6 +217,182 @@ fig.update_layout(annotations=annotations)
 fig.show()
 ```
 
````
the initial diverging bar is a paste of #4994.  Happy to roll back the change that pasted it in; if we do so, I think the example we wrote as the 2nd diverging bar example will appear first, before the simpler diverging bar example.  I can fix that with a separate, future PR.

rl-utility-man commented:
@LiamConnors Is there anything I can do to help finalize this?  I responded to all of your points a couple weeks ago, think we're substantially in consensus on the substance and very close to done.

rl-utility-man committed:
removed the paste of the first diverging bar example 

I pasted in a diverging bar example that's already in doc-prod; but now think it added more confusion than clarity

gvwilson commented:
@rl-utility-man thanks for all your hard work - I'll merge this as soon as Plotly.py 6.1 has gone up on PyPI (which should be today or tomorrow).

rl-utility-man committed:
edited introductory text to be order independent

this ensures that the introductory text makes sense even if the more complex example appears first.

rl-utility-man commented:
> @rl-utility-man thanks for all your hard work - I'll merge this as soon as Plotly.py 6.1 has gone up on PyPI (which should be today or tomorrow).

Thanks @gvwilson!  Looking forward to supporting the effort to finalize this.  I'm excited that Plotly 6.1 is out and was curious when you expect kaleido 1.0 to be released?

rl-utility-man commented:
@camdecoster I just got back to contributing to the Plotly documentation and realized that Plotly staff thought this was ready to merge a year ago, but never merged it in.  This just adds a new example, so I'd be shocked if there were conflicts with other edits.  It runs fine on plotly 6.6.  Let me know if there's anything I can do to facilitate finishing this up and getting this out to users.
